news/10903.removal.rst

```diff
@@ -0,0 +1 @@
+Drop the doctype check for the simple index protocol.
```

Comment:
IMHO you should focus on the --find-links usecase only for now.

Reply:
Well, the changelog entry is in line with what the changes made here are.

Reply:
I imagine that doing so would be harder, and no-one is that keen on keeping the check for indexes anyway. But in any case, it's not like there's a rush here, I'm assuming this is targeting 22.1 (April).

Reply:
I won't mind putting this into a final bugfix release for 22.0 -- all the index providers that needed to pay attention and make changes have likely started doing so. At this point, we're in somewhat the same camp as the root warning -- except, it's definitely not in the control of the end user what the index is doing.

I think it might make sense to bring back the warning in a less noisy form (eg: once per domain or something) later. That seems like a better UX story than burying a bunch of our other UX improvements "under" this warning. I'm biased, but I'd like more people to see the better error messages and UI improvements; instead of this warning. :P